src/org/ods/OdsContext.groovy

```diff
@@ -291,22 +315,50 @@ class OdsContext implements Context {
       config.bitbucketHost
   }
 
-  boolean getEnvironmentCreated() {
-      this.environmentCreated
-  }
-
   int getOpenshiftBuildTimeout() {
       config.openshiftBuildTimeout
   }
 
-  def setEnvironmentCreated(boolean created) {
-      this.environmentCreated = created
+  boolean getCiSkipEnabled() {
+    return config.ciSkipEnabled
+  }
+
+  void setCiSkipEnabled(boolean ciSkipEnabled) {
+    config.ciSkipEnabled = ciSkipEnabled
+  }
+
+  boolean getBitbucketNotificationEnabled() {
+    return config.bitbucketNotificationEnabled
+  }
+
+  void setBitbucketNotificationEnabled(boolean bitbucketNotificationEnabled) {
+    config.bitbucketNotificationEnabled = bitbucketNotificationEnabled
+  }
+
+  boolean getLocalCheckoutEnabled() {
+    return config.localCheckoutEnabled
+  }
+
+  void setLocalCheckoutEnabled(boolean localCheckoutEnabled) {
+    config.localCheckoutEnabled = localCheckoutEnabled
+  }
+
+  boolean getDisplayNameUpdateEnabled() {
+    return config.displayNameUpdateEnabled
+  }
+
+  void setDisplayNameUpdateEnabled(boolean displayNameUpdateEnabled) {
+    config.displayNameUpdateEnabled = displayNameUpdateEnabled
   }
 
   private String retrieveGitUrl() {
-    script.sh(
-      returnStdout: true, script: 'git config --get remote.origin.url'
-    ).trim().replace('https://bitbucket', 'https://cd_user@bitbucket')
+    //password must also be mapped, otherwise username is null
+    script.withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: this.credentialsId,
+                             usernameVariable: 'CD_USERNAME', passwordVariable: 'CD_PASSWORD']]) {
+      script.sh(
+              returnStdout: true, script: 'git config --get remote.origin.url'
+      ).trim().replaceAll('^(https?)://') { it[0] + "${script.env.CD_USERNAME}@" }
```

Comment:
call this on an existing `//cduser@` will fail .. I have fixed this in the local //97 branch now with groovy and a check if `@*  is there..